src/ProjectOrigin.PedersenCommitment/Ristretto/SigmaProof.cs

```diff
@@ -26,15 +26,18 @@ private ZeroProof(Scalar c, Scalar z)
     public static ZeroProof Prove(Generator gen, Scalar r, byte[] label)
     {
         var c0 = gen.H() * r;
-        var oracle = (Point A) => Oracle(label, A, c0, gen.G(), gen.H());
+        var oracle = new Oracle(label);
+        oracle.Add(c0);
         return ZeroProof.Prove(gen, r, oracle);
     }
 
-    internal static ZeroProof Prove(Generator gen, Scalar r, Func<Point, Scalar> oracle)
+    internal static ZeroProof Prove(Generator gen, Scalar r, Oracle oracle)
     {
         var a = Scalar.Random();
         var A = gen.H() * a;
-        var c = oracle(A);
+        oracle.Domain("ZeroProof");
+        oracle.Add(A, gen.G(), gen.H());
```

Comment:
```suggestion
//this only works if the oracle is adapted
        oracle.AddPoint("generatorG",gen.G());
        oracle.AddPoint("generatorH",gen.H());
        oracle.AddPoint("firstMessageA",A);
```

Reply:
Note that the exact label text is not that relevant (as long as labels are not prefixes of each other).

Reply:
Should the label be added as a parameter in the .Add() function to make the functionality more generic and less prone to prefixing @tschudid ?

Reply:
Yes, the add function(s) (see comments in the oracle itself) should take a label as mandatory input. The prefixing should not be a big issue (as we only hash a few messages per proof).